python/setup.py

```diff
@@ -214,37 +231,46 @@ def run(self):
                     dst_file = path.join(mac_os_static_libs_dir, lib_file_name)
                     shutil.copyfile(src_file, dst_file)
 
-            if not path.exists(path.join(tpie_lib_dir, 'libtpie.a')):
-                try:
-                    cpu_count = multiprocessing.cpu_count()
-                except:
-                    cpu_count = 1
-
-                CMAKE_CXX_FLAGS = '-fPIC -std=c++11'
-                if sys.platform == 'darwin':
-                    CMAKE_CXX_FLAGS += ' -mmacosx-version-min=10.9'
-
-                tpie_build_cmd = 'mkdir -p {}'.format(tpie_build_dir)
-                tpie_build_cmd += ' && cd {}'.format(tpie_build_dir)
-                tpie_build_cmd += ' && cmake -D CMAKE_BUILD_TYPE:STRING=Release ' \
-                                  ' -D TPIE_PARALLEL_SORT=1 -D COMPILE_TEST=OFF -D CMAKE_CXX_FLAGS="{CXX_FLAGS}"' \
-                                  ' -D CMAKE_INSTALL_PREFIX={INSTALL_PREFIX} ..'.format(
-                    CXX_FLAGS=CMAKE_CXX_FLAGS, INSTALL_PREFIX=tpie_install_prefix)
-                tpie_build_cmd += ' && make -j {}'.format(cpu_count)
-                tpie_build_cmd += ' && make install'
+            CMAKE_CXX_FLAGS = '-fPIC -std=c++11'
+            if sys.platform == 'darwin':
+                CMAKE_CXX_FLAGS += ' -mmacosx-version-min=10.9'
+
+            keyvi_build_cmd = 'mkdir -p {}'.format(keyvi_build_dir)
+            keyvi_build_cmd += ' && cd {}'.format(keyvi_build_dir)
+            keyvi_build_cmd += ' && cmake -D CMAKE_BUILD_TYPE:STRING=python ' \
+                                ' -D CMAKE_CXX_FLAGS="{CXX_FLAGS}"' \
+                                ' -D CMAKE_INSTALL_PREFIX={INSTALL_PREFIX}'.format(
+                CXX_FLAGS=CMAKE_CXX_FLAGS, INSTALL_PREFIX=keyvi_install_prefix)
+            if zlib_root is not None:
+                 keyvi_build_cmd += ' -D ZLIB_ROOT={ZLIB_ROOT}'.format(ZLIB_ROOT=zlib_root)
+            keyvi_build_cmd += ' ..'
+            keyvi_build_cmd += ' && make -j {}'.format(cpu_count)
+            keyvi_build_cmd += ' && make install'
+
+            print ("Building keyvi C++ part: " + keyvi_build_cmd)
+            subprocess.call(keyvi_build_cmd, shell=True)
```

Comment:
this is building keyvi now instead of only tpie, the reason is the keyvimerger binary